https-proxy-agent.js

```diff
@@ -36,6 +36,9 @@ function HttpsProxyAgent (opts) {
   // if `true`, then connect to the proxy server over TLS. defaults to `false`.
   this.secureProxy = proxy.protocol ? /^https:?$/i.test(proxy.protocol) : false;
 
+  // use custom CA certificate
+  this.ca = opts.ca;
```

Comment:
@TooTallNate Have you been able to give this any thought? 

Passing `ca` into `node-fetch-npm` (as **npm@5** does) had no effect, it needed to be on the agent's (this dependency) `tls.connect()` call. Happy to take your advice if you want anything tweaking.